### ISSUE TITLE
Clang 3.4 modules test fails at run-time

### ISSUE THREAD

**Issue by AlexanderLanin** [ccache/ccache#714]
### How to reproduce ###

Either compile with a new compiler and run tests on clang-3.4 or execute `VERBOSE=1 CC=clang CXX=clang++ CMAKE_PARAMS="-DZSTD_FROM_INTERNET=ON" misc/build-in-docker ubuntu-14.04` after merging several of the currently open pull requests.

### Actual behavior ###
```
Test suite:     modules
Test case:      cache miss
Failure reason: Expected "cache miss" to be 3, actual 2

ccache -s:
cache directory                     /source/builddir/testdir/12600/.ccache
primary config                      /source/builddir/testdir/12600/.ccache/ccache.conf
secondary config (readonly)         
stats updated                       Fri Oct 30 21:20:07 2020
cache hit (direct)                     1
cache hit (preprocessed)               0
cache miss                             2
cache hit rate                     33.33 %
called for link                        2
no input file                          2
cleanups performed                     0
files in cache                         3
cache size                          12.3 kB
max cache size                       5.0 GB
```
[failed.tar.gz](https://github.com/ccache/ccache/files/5467733/failed.tar.gz)

### Expected behavior ###
Pass tests (either proper support or always call real compiler directly)

### Environment ###
clang-3.4
current ccache master

**Comment by jrosdahl** [comment]
I'm tempted to close this without action. What do you think, @AlexanderLanin?

**Comment by AlexanderLanin** [comment]
Hi, sorry my contributions have been limited in recent times.
It seems unreasonable to introduce fixes for clang 3.4.
It's even unreasonable to investigate it.
And we have since then actively excluded clang 3.4.

Optionally add a reference to this ticket:
https://github.com/ccache/ccache/blob/master/CMakeLists.txt#L31

Otherwise I'm fine with just closing it.

**Comment by jrosdahl** [comment]
OK, thanks.
